docs/user/installation.rst

```diff
@@ -4,50 +4,239 @@
+
+**Optional**: If you plan to contribute to Lasagne, you will need to fork the
+Lasagne repository on GitHub. This will create a repository under your user
+account. Update your local clone to refer to the official repository as
+``upstream``, and your personal fork as ``origin``:
+
+.. code-block:: bash
+
+  git remote rename origin upstream
+  git remote add origin https://github.com/<your-github-name>/Lasagne.git
+
+If you set up an `SSH key <https://help.github.com/categories/ssh/>`_, use the
+SSH clone URL instead: ``git@github.com:<your-github-name>/Lasagne.git``.
+
+You can now use this installation to develop features and send us pull requests
+on GitHub, see :doc:`development`!
+
+  
+Installation without pip
+========================
+
+If, for some reason, you need to install Theano and Lasagne without ``pip``,
+you will need to obtain the source trees:
+
+.. code-block:: bash
+
+  git clone https://github.com/Theano/Theano.git
+  git clone https://github.com/Lasagne/Lasagne.git
+
+And then install via ``setup.py``:
 
 .. code-block:: bash
 
-  python setup.py install  # or 'setup.py develop' if you're developing Lasagne itself
+  cd Theano && python setup.py install && cd -
+  cd Lasagne && python setup.py install && cd -
+
+Append ``--prefix=~/.local`` after ``setup.py`` to install to your home
+directory, and replace ``install`` with ``develop`` to install in editable
+mode. Note that this will install the bleeding-edge versions of Theano and
+Lasagne.
+
+
+GPU support
+===========
+
+Thanks to Theano, Lasagne transparently supports training your networks on a
+GPU, which may be 10 to 50 times faster than training it on a CPU. Currently,
+this requires an NVIDIA GPU with CUDA support, and some additional software for
+Theano to use it.
+
+CUDA
+----
+
+Install the latest CUDA Toolkit and possibly the corresponding driver available
+from NVIDIA: https://developer.nvidia.com/cuda-downloads
+
+Closely follow the *Getting Started Guide* linked underneath the download table
+to be sure you don't mess up your system by installing conflicting drivers.
+
+After installation, make sure ``/usr/local/cuda/bin`` is in your ``PATH``, so
+``nvcc --version`` works. Also make sure ``/usr/local/cuda/lib64`` is in your
+``LD_LIBRARY_PATH``, so the toolkit libraries can be found.
+
+Theano
+------
+
+If CUDA is set up correctly, the following should print some information on
+your GPU (the first CUDA-capable GPU in your system if you have multiple ones):
+
+.. code-block:: bash
+
+  THEANO_FLAGS=device=gpu python -c "import theano; print theano.sandbox.cuda.device_properties(0)"
+
+To configure Theano to use the GPU by default, create a file ``.theanorc``
+directly in your home directory, with the following contents:
+
+.. code-block:: none
+
+  [global]
+  floatX = float32
+  device = gpu
+
+Optionally add ``allow_gc = False`` for some extra performance at the expense
+of some more GPU memory.
```

Comment:
maybe "at the expensive of higher GPU memory usage". The way it's phrased right now makes it sound like there might be a modest increase, but it can actually be quite substantial :)